config/secure-headers.php

```diff
@@ -316,47 +316,227 @@
 	/*
 	 * There is no easy way to use CSP with debug bar at the moment so we disable CSP if debug bar is enabled.
 	 */
-	'custom-csp' => ((bool) env('DEBUGBAR_ENABLED', false)) ? '' : null,
-
 	'csp' => [
+		'enable' => ((bool) env('DEBUGBAR_ENABLED', false)) === false,
+
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy-Report-Only
 		'report-only' => false,
-		'report-uri' => null,
+
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/report-to
+		'report-to' => '',
+
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/report-uri
+		'report-uri' => [
+			// uri
+		],
+
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/block-all-mixed-content
 		'block-all-mixed-content' => false,
+
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/upgrade-insecure-requests
 		'upgrade-insecure-requests' => false,
 
-		/*
-		 * Please references script-src directive for available values, only `script-src` and `style-src`
-		 * supports `add-generated-nonce`.
-		 *
-		 * Note: when directive value is empty, it will use `none` for that directive.
-		 */
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/base-uri
+		'base-uri' => [
+		],
+
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/child-src
+		'child-src' => [
+		],
+
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/connect-src
+		'connect-src' => [
+			'https://lycheeorg.github.io/update.json',
+		],
+
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/default-src
+		'default-src' => [
+		],
+
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/font-src
+		'font-src' => [
+		],
+
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/form-action
+		'form-action' => [
+			'self' => true,
+		],
+
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/frame-ancestors
+		'frame-ancestors' => [
+		],
+
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/frame-src
+		'frame-src' => [
+		],
+
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/img-src
+		'img-src' => [
+		],
+
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/manifest-src
+		'manifest-src' => [
+		],
+
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/media-src
+		'media-src' => [
+			'self' => true,
+			'allow' => [
+				'blob:',
```

Comment:
Do we also want `env('LYCHEE_UPLOADS_URL', 'https://lycheeorg.github.io/'),` here?

Reply:
Removed in [3f5583f](https://github.com/LycheeOrg/Lychee/pull/1528/commits/3f5583f3f4300551dbb90229ed1d7e908202272d)

Reply:
I am not sure what @d7415 wanted to imply with his first comment and am not sure whether we should _add_ `LYCHEE_UPLOADS_URL`. Consider the case that the backend is running on a different server than the upload path resides on and that the images as well as videos (in case of life photos) are hosted below a different domain. (I know, this is currently still not supported.) In this case it does not suffice to allow `'self'` as the origin for images and media.

However, I do agree that `'https://lycheeorg.github.io/'` shouldn't be the default.

Reply:
I meant that it *was* present for images (it's been removed, which I'm not convinced by, but it has) and so the same should apply for media if videos are stored there. I assume the default was set because of the demo, but I think that case should be covered by `self`.

If `LYCHEE_UPLOADS_URL` isn't allowed in either section until we add (proper) support for media on another server, I'm fine with that, but if it's in one it should be in both.

Reply:
It's currently in neither, so that sounds good enough for both of us.

Reply:
Then I resolve this one.